synapse/rest/client/devices.py

```diff
@@ -179,6 +179,12 @@ class DeleteBody(RequestBodyModel):
     async def on_DELETE(
         self, request: SynapseRequest, device_id: str
     ) -> Tuple[int, JsonDict]:
+        # Can only be run on main process
+        if not self._is_main_process:
+            logger.error("DELETE on /devices/ must be routed to main process")
+            raise SynapseError(500, "Server misconfigured")
```

Comment:
I think it's fine to tell the end user this error message. It would save the sysadmin (i.e. us) needing to dig out the logs to figure out the misconfiguration.

```suggestion
            error_message = "DELETE on /devices/ must be routed to main process"
            logger.error(error_message)
            raise SynapseError(500, error_message)
```

likewise below.